app/models/miq_template.rb

```diff
@@ -41,4 +50,74 @@ def active?; false; end
   def self.display_name(number = 1)
     n_('Template and Image', 'Templates and Images', number)
   end
+
+  def self.non_deprecated
+    where(:deprecated => false).or(where(:deprecated => nil))
+  end
+
+  def display_type
+    if respond_to?(:volume_template?)
+      _("Volume")
+    elsif respond_to?(:volume_snapshot_template?)
+      _("Volume Snapshot")
+    elsif respond_to?(:image?)
+      image? ? _("Image") : _("Snapshot")
+    else
+      _("N/A")
+    end
+  end
+
+  def display_operating_system
+    if respond_to?(:volume_template?) || respond_to?(:volume_snapshot_template?)
+      _("N/A")
+    else
+      operating_system.try(:product_name)
+    end
+  end
+
+  def display_platform
+    if respond_to?(:volume_template?) || respond_to?(:volume_snapshot_template?)
+      _("N/A")
+    else
+      platform
+    end
+  end
+
+  def display_cpu_cores
+    if respond_to?(:volume_template?) || respond_to?(:volume_snapshot_template?)
+      _("N/A")
+    else
+      cpu_total_cores
+    end
+  end
+
+  def display_memory
+    if respond_to?(:volume_template?) || respond_to?(:volume_snapshot_template?)
+      _("N/A")
+    else
+      mem_cpu.to_i * 1024 * 1024
+    end
+  end
+
+  def display_snapshots
+    if respond_to?(:volume_template?) || respond_to?(:volume_snapshot_template?)
+      _("N/A")
+    else
+      v_total_snapshots
+    end
+  end
+
+  def display_tenant
+    respond_to?(:cloud_tenant) ? cloud_tenant.try(:name) : _("N/A")
+  end
+
+  def display_deprecated
+    if respond_to?(:volume_template?) || respond_to?(:volume_snapshot_template?)
+      _("N/A")
+    elsif respond_to?(:deprecated)
+      _(deprecated.to_s)
```

Comment:
@mzazrivec is this ok?

Reply:
Yes, as long as `deprecated.to_s` is already tagged somewhere for string collection. Otherwise this won't work.